### ISSUE TITLE
Python hightlight `in` and `not in` inconsitent

### ISSUE THREAD

**Issue by tdegeus** [microsoft/vscode#194484]
Type: <b>Bug</b>

In the following snippet `not` and `in` are incorrectly hightlighted, see screenshot.

```python
import h5py

def Run():
    with h5py.File("foo", "a") as file:
        if "a" not in file:
            assert not any(m in file for m in ["b"]), "Wrong file type"
        else:
            pass
```

Instead, this correct hightlighted is given if I (incorrectly) remove the import:
```python
def Run():
    with h5py.File("foo", "a") as file:
        if "a" not in file:
            assert not any(m in file for m in ["b"]), "Wrong file type"
        else:
            pass
```

VS Code version: Code 1.82.2 (Universal) (abd2f3db4bdb28f9e95536dfa84d8479f1eb312d, 2023-09-14T05:59:47.790Z)
OS version: Darwin arm64 22.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|16.00GB (0.08GB free)|
|Process Argv|--crash-reporter-id 3d97051d-96cb-46f2-aa74-cf444e361b1f|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|1.4.0
antidote|dru|1.0.21
pre-commit-helper|ela|0.4.3
vscode-diff|fab|1.4.2
vscode-sublime-merge|gio|1.4.0
copilot|Git|1.121.453
copilot-chat|Git|0.7.1
vscode-github-actions|git|0.26.2
todo-tree|Gru|0.0.226
latex-workshop|Jam|9.14.0
better-cpp-syntax|jef|1.17.2
isort|ms-|2023.10.1
python|ms-|2023.16.0
vscode-pylance|ms-|2023.9.30
jupyter|ms-|2023.8.1002501831
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-ssh|ms-|0.106.4
remote-ssh-edit|ms-|0.86.0
cmake-tools|ms-|1.15.31
cpptools|ms-|1.17.5
cpptools-extension-pack|ms-|1.3.0
remote-explorer|ms-|0.4.1
vs-sequential-number|nep|1.1.0
code-spell-checker|str|3.0.1
cmake|twx|0.0.17
clang-format|xav|1.9.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscaac:30438847
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
282f8724:30602487
f6dab269:30613381
showlangstatbar:30737416
03d35959:30757346
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxptcf:30805731
pythonnoceb:30805159
copilotsettingt:30839829
dsvsc013:30795093
dsvsc014:30804076
diffeditorv1:30821571
pythonmpsinfo:30842941
dsvsc015:30845448

```

</details>

**Comment by tdegeus** [comment]
I tried "Disable all extensions for the current workspace" and the error is still appearing. Am I debugging this correctly? Are we on the same version?

**Comment by alexr00** [comment]
Yes, same version. Did you reload after disabling all extensions?

**Comment by tdegeus** [comment]
No I did not. The error disappears upon reloading! So we just have to figure out which extension causes the problem.
